test/math.jl

```diff
@@ -678,3 +678,20 @@ let x = 2.0
     @test exp2(Float22716(x)) === 2^x
     @test exp10(Float22716(x)) === 10^x
 end
+
+@testset "asin #23088" begin
+    for T in (Float32, Float64)
+        @test asin(zero(T)) == zero(T)
+        @test asin(-zero(T)) == -zero(T)
+        @test asin(nextfloat(zero(T))) == nextfloat(zero(T))
+        @test asin(prevfloat(zero(T))) == prevfloat(zero(T))
+        @test asin(one(T)) == T(pi)/2
+        @test asin(-one(T)) == -T(pi)/2
+        for x in (0.45, 0.6, 0.98)
+            by = T(asin(big(x)))
```

Comment:
Strictly speaking, the conversion to `BigFloat` in this way is unsafe, the best way to initialize a `BigFloat` is by using strings.  E.g.:
```julia
julia> big(0.45)
4,500000000000000111022302462515654042363166809082031250000000000000000000000000e-01

julia> big"0.45"
4,499999999999999999999999999999999999999999999999999999999999999999999999999983e-01

julia> Float64(asin(big(0.45)))
0.4667653390472964

julia> Float64(asin(big"0.45"))
0.46676533904729633
```
Anyway, for the precision required this may not be an issue.